Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/keydownload/ui/KeyFileDownloadAdapter.kt

```diff
@@ -0,0 +1,70 @@
+package de.rki.coronawarnapp.test.keydownload.ui
+
+import android.text.format.Formatter
+import android.view.ViewGroup
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.databinding.FragmentTestKeydownloadAdapterLineBinding
+import de.rki.coronawarnapp.diagnosiskeys.storage.CachedKeyInfo
+import de.rki.coronawarnapp.ui.lists.BaseAdapter
+import de.rki.coronawarnapp.util.lists.BindableVH
+import de.rki.coronawarnapp.util.lists.diffutil.AsyncDiffUtilAdapter
+import de.rki.coronawarnapp.util.lists.diffutil.AsyncDiffer
+import de.rki.coronawarnapp.util.ui.setGone
+import org.joda.time.format.DateTimeFormat
+
+class KeyFileDownloadAdapter(
+    private val deleteAction: (CachedKeyListItem) -> Unit
+) : BaseAdapter<KeyFileDownloadAdapter.VH>(), AsyncDiffUtilAdapter<CachedKeyListItem> {
+
+    init {
+        setHasStableIds(true)
+    }
+
+    override val asyncDiffer: AsyncDiffer<CachedKeyListItem> = AsyncDiffer(this)
+
+    override fun getItemCount(): Int = data.size
+
+    override fun getItemId(position: Int): Long = data[position].stableId
+
+    override fun onCreateBaseVH(parent: ViewGroup, viewType: Int): VH = VH(parent)
+
+    override fun onBindBaseVH(holder: VH, position: Int) {
+        val item = data[position]
+        holder.itemView.setOnLongClickListener {
+            deleteAction(item)
+            true
+        }
+        holder.bind(item)
+    }
+
+    class VH(
+        val parent: ViewGroup
+    ) : BaseAdapter.VH(
+        R.layout.fragment_test_keydownload_adapter_line, parent
+    ), BindableVH<CachedKeyListItem, FragmentTestKeydownloadAdapterLineBinding> {
+
+        override val viewBinding = lazy { FragmentTestKeydownloadAdapterLineBinding.bind(itemView) }
+
+        override val onBindData: FragmentTestKeydownloadAdapterLineBinding.(key: CachedKeyListItem) -> Unit = { item ->
+            locationInfo.text = item.info.location.identifier
+
+            val shortSize = Formatter.formatShortFileSize(context, item.fileSize)
+            typeInfo.text = when (item.info.type) {
```

Comment:
minor: the when cases could be moved to dedicated "formatter" functions

Reply:
Yes, but for test only code this is clean enough :relaxed: